allennlp/modules/seq2seq_encoders/multi_head_self_attention.py

```diff
@@ -60,12 +60,14 @@ def __init__(self,
             raise ValueError(f"Value size ({values_dim}) must be divisible by the number of "
                              f"attention heads ({num_heads}).")
 
-
         self._combined_projection = Linear(input_dim, 2 * attention_dim + values_dim)
 
         self._scale = (input_dim // num_heads) ** 0.5
         self._output_projection = Linear(values_dim, self._output_dim)
-        self._attention_dropout = Dropout(attention_dropout_prob)
+        if attention_dropout_prob > 0:
```

Comment:
What did you decide here?  This uses less memory if you specify no dropout?

Reply:
I ran an experiment just now to compare `lambda x: x`, `dropout(0)` and `dropout(0.5)`. It turns out that `dropout(0)` will not consume more memory than `lambda x: x`. So, I will change this back.